functions/lib/stripe/checkout.py

```diff
@@ -0,0 +1,157 @@
+      transaction.update(organiser_ref, {"stripeAccountActive": True})
+      logger.info(f"Provided organiser {organiser_ref.path} already has all charges enabled and details submitted. Activiating their sportshub stripe account.")
+
+    else:
+      logger.error(f"Provided event {event_ref.path} has an organiser {organiser_ref.path} who does not have a active stripe account. charges_enabled={account.charges_enabled} details_submitted={account.details_submitted} Returning status=500")
+      return json.dumps({"url": ERROR_URL})
+
+  # 3. check again with database in transction in the backend if quantity is still available
+  vacancy = event.get("vacancy")
+
+  if vacancy < quantity:
+    logger.warning(f"Provided event {event_ref.path} does not have enough tickets to fulfill this order. quantity_requested={quantity} vacancy={vacancy}")
+    return json.dumps({"url": cancel_url})
+  
+  # 4. obtain price and organiser stripe account id again from db
+  price = event.get("price")
+  organiser_stripe_account_id = organiser.get("stripeAccount")
+
+  # 4a. check if the price exists for this event
+  if (price == None or not isinstance(price, int) or price <= 1): # we don't want events to be less than stripe fees
+    logger.error(f"Provided event {event_ref.path} does not have a valid price. Returning status=500")
+    return json.dumps({"url": ERROR_URL})
+
+  # 5. set the tickets as sold and reduce vacancy (prevent race condition/ over selling, we will release tickets back after cancelled sale)
+  transaction.update(event_ref, {"vacancy": vacancy - quantity })
+  logger.info(f"Securing {quantity} tickets for event {event_ref.path} at ${price}. There are now {vacancy - quantity} tickets left.")
+
+  # 6. create checkout session with connected account and return link
+  checkout = stripe.checkout.Session.create(
+    mode="payment",
+    line_items=[{
+      "price_data": {
+        "currency": "aud",
+        "product_data": {
+          "name": event.get("name"),
+          "metadata": {
+            "eventId": event_id,
+            "isPrivate": is_private
+          }
+        },
+        "unit_amount": price * 100
+      },
+      "quantity": quantity
+    }],
+    metadata={
+      "eventId": event_id,
+      "isPrivate": is_private
+    },
+    # payment_intent_data={"application_fee_amount": 123},
+    success_url="https://example.com/success",
+    cancel_url=cancel_url,
+    stripe_account= organiser_stripe_account_id,
+    expires_at=int(time.time() + 1800) # Checkout session expires in 5 minutes
+  )
+
+  logger.info(f"Creating checkout session {checkout.id} for event {event_ref.path}, linked to {organiser_ref.path} and their stripe account {organiser_stripe_account_id}. Secured {quantity} tickets at ${price}.")
+  return json.dumps({"url": checkout.url})
+
+
+@https_fn.on_call(cors=options.CorsOptions(cors_origins=["localhost", "www.sportshub.net.au", "*"], cors_methods=["post"]), region="australia-southeast1")
```

Comment:
should we replace the "*" wildcard with a regex that matches "social-sport-*" to match the deployments

Reply:
so i had alot of trouble with this for some reason (CORS errors), im going to keep this as is for now with the wildcard as I don't want that to delay merging in this stripe PR. I will make a new ticket under the stripe epic to have a task open to fix this issue

Reply:
https://owenyang.atlassian.net/jira/software/c/projects/SPORTSHUB/boards/6?selectedIssue=SPORTSHUB-79